tests/components/sensor/test_pilight.py

```diff
@@ -0,0 +1,117 @@
+"""The tests for the Pilight sensor platform."""
+import unittest
+from unittest.mock import patch
+
+from homeassistant.bootstrap import _setup_component
+import homeassistant.components.sensor as sensor
+from homeassistant.components import pilight
+
+from tests.common import get_test_home_assistant
+
+
+def fire_pilight_message(hass, protocol, data):
+    """Fire the fake pilight message."""
+    message = {pilight.ATTR_PROTOCOL: protocol}
+    message.update(data)
+    hass.bus.fire(pilight.EVENT, message)
+
+
+class TestSensorPilight(unittest.TestCase):
+    """Test the Pilight sensor."""
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+        self.hass.config.components = ['pilight']
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop down everything that was started."""
+        self.hass.stop()
+
+    def test_sensor_value_from_code(self):
+        """Test the setting of value via pilight."""
+        assert _setup_component(self.hass, sensor.DOMAIN, {
```

Comment:
Please use `setup_component` and wrap it in `assert_setup_component`.

The description of #3736 explains how and why.